runtime-k3d/src/gen_podspec.py

```diff
@@ -0,0 +1,129 @@
+    
+    container_image = None
+    service_port = None
+    env_vars = dict[str, Optional[str]]()
+    port_forwards = []
+    mounts = []
+    args = []
+    pods=[]
+
+    for config_entry in service_spec["config"]:
+        if config_entry["key"] == "image":
+            container_image = config_entry["value"]
+        elif config_entry["key"] == "env":
+            pair = config_entry["value"].split("=", 1)
+            env_vars[pair[0].strip()] = None
+            if len(pair) > 1:
+                env_vars[pair[0].strip()] = pair[1].strip()
+        elif config_entry["key"] == "port":
+            service_port = config_entry["value"]
+        elif config_entry["key"] == "no-dapr":
+            no_dapr = config_entry["value"]
+        elif config_entry["key"] == "arg":
+            args.append(config_entry["value"])
+        elif config_entry["key"] == "port-forward":
+            port_forwards.append(config_entry["value"])
+        elif config_entry["key"] == "mount":
+            mounts.append(config_entry["value"])
+    
+    template_pod = templates[find_service_spec(templates, "Pod", service_id)[0]]
+    template_pod['spec']['containers'][0]['image'] = container_image
+    if args:
+        template_pod['spec']['containers'][0]['args'] = '[ ' + ', '.join(f'"{arg}"' for arg in args) + ' ]'
+    if env_vars:
+        template_pod["spec"]["containers"][0]["env"] = []
+        for key, value in env_vars.items():
+            template_pod["spec"]["containers"][0]["env"].append({"name": key,
+                                                                 "value": value})
+
+    pods.append(template_pod)
+    if service_port:
+        template_pod["spec"]["containers"][0]['ports'] = [{'name': 'default',
+                                                           'containerPort': int(service_port),
+                                                           'protocol': 'TCP'}]
+    
+    if port_forwards:
+        nodeport_spec = create_nodeport(service_id)
+        nodeport_spec["spec"]['ports'] = []        
+        for port in port_forwards:
+            source_target_ports = port.split(':')
+            nodeport_spec["spec"]['ports'].append({
+                "port": int(source_target_ports[0]),
+                "targetPort": int(source_target_ports[1]),
+                "nodePort": node_port
+            })
+            node_port = node_port + 1
+        pods.append(nodeport_spec)
+                
+    # TBD: mounts
+    
+    return pods
+
+if __name__ == "__main__":
+    with open(f"{get_script_path()}/runtime/config/podspec/runtime_template.yaml", 'r') as f:
+        templates = list(yaml.load_all(f, Loader=SafeLoader))
+        
+    pods = init_template(templates)
+    for service in get_services():
+        pods.extend(create_podspec(templates, service_spec=service))
+        
+    with open(f"{get_script_path()}/runtime/config/podspec/runtime_generated.yaml", 'w') as f:
+        yaml.dump_all(pods, f)
```

Comment:
There is no newline at the end of this file, but in addition there are some more small "formatting & linter" complains throughout the file. Some line too long or even some trailing whitespaces